tests/unit/test_utils.py

```diff
@@ -971,3 +971,29 @@ def test_make_setuptools_shim_args__unbuffered_output(unbuffered_output):
         unbuffered_output=unbuffered_output
     )
     assert ('-u' in args) == unbuffered_output
+
+
+def mock_stdin_isatty(monkeypatch, return_value):
+    monkeypatch.setattr(sys.stdin, 'isatty', Mock(return_value=return_value))
+
+
+def test_is_console_interactive_when_stdin_is_none_and_isatty_true(monkeypatch):
```

Comment:
Quick note, if you want to cover all scenarios succinctly, check out [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) - here we could do something like

```
@pyest.mark.parametrize('isatty,no_stdin,expected', [
    (True, False, True),
    ...
])
def test_is_console_interactive(isatty, no_stdin, expected):
    ...
```

and have a function that would work similarly to the below. If it doesn't look too unreadable, we may also inline `mock_stdin_isatty` since it would all be in one place.

Reply:
Wow, if I got the usage right, that's a lot more succint/readable/nice.

Reply:
Yup, looks good to me. :)